Z/ZPares/build_tarballs.jl

```diff
@@ -0,0 +1,103 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "ZPares"
+version = v"0.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/cometscome/zpares_mirror.git", "ab453f5c3aa295bb43f6396e796db8d03964b5b3")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd zpares_mirror/
+cd originalfile/
+tar -xvf zpares_0.9.6a.tar.gz 
+cd zpares_0.9.6a
+cp ../../wrapper/zpares_wrapper.f90 ./
+
+if [[ $target == *"aarch64-apple-darwin"* ]]; then 
+    Rankmismatch="-fallow-argument-mismatch"
+elif [[ $target == *"aarch64-linux"* ]]; then 
+    Rankmismatch="" 
+else
+    Rankmismatch=""
+fi
+
+
+if [[ $target == *"apple-darwin"* ]]; then
+    cp Makefile.inc/make.inc.gfortran.seq ./make.inc
+    make BLAS="-L$libdir -lopenblas" USE_MPI="0" FFLAG="-O3 $Rankmismatch -dynamiclib -L$libdir -lopenblas" LAPACK="-L./"
+    gfortran -O3 -dynamiclib -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares -L$libdir -lopenblas -o libzpares.dylib
+    cp libzpares.dylib lib/
+    cp lib/libzpares.dylib $prefix/lib/
+    cp include/zpares.mod $prefix/include/
+    cp zpares_wrapper.mod $prefix/include/
+    cp lib/libzpares.a $prefix/lib/
+elif [[ $target == *"w64-mingw32"* ]]; then 
+    cp Makefile.inc/make.inc.gfortran.seq ./make.inc
+    make BLAS="-L$libdir -lopenblas" LAPACK="-L$libdir -lopenblas" FFLAG="-O3 $Rankmismatch -shared -fPIC"
+    gfortran -O3 -shared -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares  -L$libdir -lopenblas -o zpares_wrapper.a
+    ld -shared -o zpares.so --whole-archive zpares_wrapper.a 
+    cp lib/* $prefix/bin/
+    cp zpares.so lib/
+    cp lib/zpares.so $prefix/lib/libzpares.dll
+    cp include/zpares.mod $prefix/include/
+    cp zpares_wrapper.mod $prefix/include/
+    cp lib/libzpares.a $prefix/lib/
+else
+    cp Makefile.inc/make.inc.gfortran.seq ./make.inc
+    make BLAS="-L$libdir -lopenblas" LAPACK="-L$libdir -lopenblas" FFLAG="-O3 $Rankmismatch -shared -fPIC"
+    gfortran -O3 -shared -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares  -o zpares_wrapper.a
+    ld -shared -o zpares.so --whole-archive zpares_wrapper.a 
+    cp zpares_wrapper.a $prefix/lib/
+    cp zpares.so lib/
+    cp lib/zpares.so $prefix/lib/libzpares.so
+    cp include/zpares.mod $prefix/include/
+    cp zpares_wrapper.mod $prefix/include/
+    cp lib/libzpares.a $prefix/lib/
+fi
```

Comment:
At a quick glance I can't really tell what's the difference between all these branches.  Why don't you use just one build for all platforms?  Also, we usually avoid building and/or shipping static archives, they aren't much useful for use in Julia (if you plan to use this library from Julia) and they're just large files sitting on disk.

Reply:
Yes. You are right. I did not want to use static archives. But the original code generates static archives in their makefile. So I tried to use zpares.so as dynamic one. 
At line 61, I added "cp lib/libzpares.a $prefix/lib/" to avoid failure in x86_64-linux-gnu system. In other systems, we do not need libzpares.a. I think I do not understand how shared libraries work. When I understand it more, I would like to change the build_tarballs.jl to remove static archives.

Reply:
I'd still like to reduce complexity here and remove the `if`.

Reply:
Like, wouldn't
```suggestion
cp Makefile.inc/make.inc.gfortran.seq ./make.inc
make BLAS="-L${libdir} -lopenblas" USE_MPI="0" FFLAG="-O3 ${Rankmismatch} -shared -L${libdir} -lopenblas" LAPACK="-L./"
gfortran -O3 -shared -fPIC zpares_wrapper.f90 -I./include -L./lib -lzpares -L${libdir} -lopenblas -o "${libdir}/libzpares.${dlext}"
cp include/zpares.mod "${includedir}"
cp zpares_wrapper.mod "${includedir}"
```
work everywhere? Or am I missing something?

Reply:
Thank you for your suggestion. I revised this little bit. And finally all checks are passed.